### PR TITLE
659: Update Spring.Boot to 2.0.0.RELEASE

### LINKED ISSUE #659: Update Spring.Boot to 2.0.0
Spring.Boot 2.0.0 is out 🎉 

### Steps:
1. Upgrade to Spring.Boot 2.0.0
2. Bump dependencies
3. Make it working

### PR DESCRIPTION
**Pull request by netme**
Closes #659 

### Critical changes:
- Removes old-style `dropwizard` metrics
- Removes `zmon` support (doesn't support Spring Boot 2.0 yet) 
- Few changes around `application.properties`

### TODO:
- [x] Fix tests
- [x] Fix custom metrics
- [x] Bump Kotlin versions
- [x] Check which dependencies can be updated

### CODE REVIEW

**Comment by netme** [comment]
👎

**Comment by netme** [comment]
👎

**Comment by netme** [comment]
Removing `[WIP]` tag. 

@maxim-tschumak I might need your support with the deployment. It might break few things around monitoring and security.

File: server/src/main/java/de/zalando/zally/apireview/ApiViolationsController.java [hunk 1]
```diff
@@ -31,32 +32,34 @@
...
                                    ServerMessageService serverMessageService,
                                    RulesPolicy configPolicy) {
+
         this.rulesValidator = rulesValidator;
-        this.metricServices = metricServices;
         this.apiDefinitionReader = apiDefinitionReader;
         this.apiReviewRepository = apiReviewRepository;
         this.serverMessageService = serverMessageService;
         this.configPolicy = configPolicy;
+        this.apiReviewsRequested = Metrics.gauge("meter.api-reviews.requested", new LongAdder());
```
Comment (maxim-tschumak) [inline server/src/main/java/de/zalando/zally/apireview/ApiViolationsController.java]: We don't use those in-memory metrics in production anymore (switched to persistent `ApiReviewStatistics` and `/review-statistics` endpoint).

@roxspring Do you use these metrics in your setup?

If the answer is no - I propose to delete these metrics.

Comment (roxspring) [reply server/src/main/java/de/zalando/zally/apireview/ApiViolationsController.java]: No. Probably ought to be looking at them and drawing some conclusions but haven't been doing so yet.

Comment (maxim-tschumak) [reply server/src/main/java/de/zalando/zally/apireview/ApiViolationsController.java]: `/review-statistics` substitutes all the functionalities of these in-memory metrics.

@netme Please remove them

Comment (netme) [reply server/src/main/java/de/zalando/zally/apireview/ApiViolationsController.java]: @maxim-tschumak , ah, I see. Sure, will remove them

Comment (netme) [reply server/src/main/java/de/zalando/zally/apireview/ApiViolationsController.java]: @maxim-tschumak done

**Comment by netme** [comment]
👍

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by netme** [comment]
👍

**Comment by netme** [comment]
Let's ship it

**Comment by maxim-tschumak** [comment]
I'll monitor the rollout.

**Comment by netme** [comment]
Thanks @maxim-tschumak

**Comment by maxim-tschumak** [comment]
@netme there is a small build problem, please review the fix: https://github.com/zalando/zally/pull/675

**Pull request merged**